In [4]:
import pandas as pd

In [5]:
data = pd.read_csv('Credit_Card_Dataset_for_Clustering.csv')

In [11]:
# Veri setinin ilk 5 satırını göster
data.head()

CUST_ID      BALANCE  BALANCE_FREQUENCY  PURCHASES  ONEOFF_PURCHASES  \
0  C10001    40.900749           0.818182      95.40              0.00   
1  C10002  3202.467416           0.909091       0.00              0.00   
2  C10003  2495.148862           1.000000     773.17            773.17   
3  C10004  1666.670542           0.636364    1499.00           1499.00   
4  C10005   817.714335           1.000000      16.00             16.00   

   INSTALLMENTS_PURCHASES  CASH_ADVANCE  PURCHASES_FREQUENCY  \
0                    95.4      0.000000             0.166667   
1                     0.0   6442.945483             0.000000   
2                     0.0      0.000000             1.000000   
3                     0.0    205.788017             0.083333   
4                     0.0      0.000000             0.083333   

   ONEOFF_PURCHASES_FREQUENCY  PURCHASES_INSTALLMENTS_FREQUENCY  \
0                    0.000000                          0.083333   
1                    0.000000                          0.000000   
2                    1.000000                          0.000000   
3                    0.083333                          0.000000   
4                    0.083333                          0.000000   

   CASH_ADVANCE_FREQUENCY  CASH_ADVANCE_TRX  PURCHASES_TRX  CREDIT_LIMIT  \
0                0.000000                 0              2        1000.0   
1                0.250000                 4              0        7000.0   
2                0.000000                 0             12        7500.0   
3                0.083333                 1              1        7500.0   
4                0.000000                 0              1        1200.0   

      PAYMENTS  MINIMUM_PAYMENTS  PRC_FULL_PAYMENT  TENURE  
0   201.802084        139.509787          0.000000      12  
1  4103.032597       1072.340217          0.222222      12  
2   622.066742        627.284787          0.000000      12  
3     0.000000               NaN          0.000000      12  
4   678.334763        244.791237          0.000000      12

Bu tablo, kredi kartı kullanıcılarının çeşitli finansal verilerini içermektedir. Her bir satır, tek bir müşteriye ait verileri temsil eder. Sütunların anlamları ve bazı yorumlar şu şekildedir:

1. **CUST_ID**: Müşteri kimlik numarasıdır; her satır benzersiz bir müşteriyi ifade eder.
2. **BALANCE**: Müşterinin kredi kartı bakiyesidir (TL cinsinden).
3. **BALANCE_FREQUENCY**: Müşterinin bakiyesini ne sıklıkla güncel tuttuğunu ifade eden bir oran (0 - 1 arası).
4. **PURCHASES**: Müşterinin toplam alışveriş harcamasını gösterir (TL cinsinden).
5. **ONEOFF_PURCHASES**: Tek seferde yapılan alışverişler için harcanan toplam miktar.
6. **INSTALLMENTS_PURCHASES**: Taksitli alışverişler için yapılan toplam harcama.
7. **CASH_ADVANCE**: Kredi kartından çekilen nakit avans tutarı.
8. **PURCHASES_FREQUENCY**: Müşterinin alışveriş yapma sıklığını ifade eden bir oran (0 - 1 arası).
9. **ONEOFF_PURCHASES_FREQUENCY**: Tek seferlik alışverişlerin yapılma sıklığını gösteren oran.
10. **PURCHASES_INSTALLMENTS_FREQUENCY**: Taksitli alışverişlerin yapılma sıklığını ifade eden oran.
11. **CASH_ADVANCE_FREQUENCY**: Nakit avans çekme sıklığını gösteren oran.
12. **CASH_ADVANCE_TRX**: Müşterinin toplam nakit avans çekim işlemi sayısı.
13. **PURCHASES_TRX**: Müşterinin toplam alışveriş işlem sayısı.
14. **CREDIT_LIMIT**: Müşterinin kredi kartı limiti.
15. **PAYMENTS**: Müşterinin yaptığı toplam ödeme miktarı.
16. **MINIMUM_PAYMENTS**: Müşterinin ödemesi gereken minimum tutar.
17. **PRC_FULL_PAYMENT**: Müşterinin toplam borcun tamamını ödeme oranı.
18. **TENURE**: Kredi kartının kaç aydır açık olduğunu belirtir.

### Genel Gözlemler
- İlk müşterinin (C10001), **düşük bir kredi kartı limiti** var (1000 TL) ve **düzenli taksitli alışveriş** yapıyor; **nakit avans çekmemiş** ve **tam ödeme yapmıyor**.
- İkinci müşteri (C10002), **yüksek bir nakit avans çekimi** (6442.95 TL) gerçekleştirmiş, alışveriş yapmamış ve ödeme oranı nispeten düşük (0.222).
- Üçüncü müşteri (C10003), **yüksek miktarda tek seferlik alışveriş** (773.17 TL) yapmış, **nakit avans çekmemiş** ve **yüksek kredi kartı limiti** bulunuyor (7500 TL).
- Bazı müşterilerin **minimum ödeme tutarları eksik (NaN)** durumda; bu durum veri eksikliği anlamına gelir.
  
Bu veriler, müşteri segmentasyonu, harcama alışkanlıkları veya risk analizleri gibi analizlerde kullanılabilir. Özellikle **CASH_ADVANCE**, **PURCHASES_FREQUENCY** ve **PRC_FULL_PAYMENT** sütunları, müşterilerin ödeme alışkanlıkları ve nakit kullanımları hakkında değerli bilgiler sağlar.

In [13]:
# Temel istatistiksel bilgiler
data.describe()

BALANCE  BALANCE_FREQUENCY     PURCHASES  ONEOFF_PURCHASES  \
count   8950.000000        8950.000000   8950.000000       8950.000000   
mean    1564.474828           0.877271   1003.204834        592.437371   
std     2081.531879           0.236904   2136.634782       1659.887917   
min        0.000000           0.000000      0.000000          0.000000   
25%      128.281915           0.888889     39.635000          0.000000   
50%      873.385231           1.000000    361.280000         38.000000   
75%     2054.140036           1.000000   1110.130000        577.405000   
max    19043.138560           1.000000  49039.570000      40761.250000   

       INSTALLMENTS_PURCHASES  CASH_ADVANCE  PURCHASES_FREQUENCY  \
count             8950.000000   8950.000000          8950.000000   
mean               411.067645    978.871112             0.490351   
std                904.338115   2097.163877             0.401371   
min                  0.000000      0.000000             0.000000   
25%                  0.000000      0.000000             0.083333   
50%                 89.000000      0.000000             0.500000   
75%                468.637500   1113.821139             0.916667   
max              22500.000000  47137.211760             1.000000   

       ONEOFF_PURCHASES_FREQUENCY  PURCHASES_INSTALLMENTS_FREQUENCY  \
count                 8950.000000                       8950.000000   
mean                     0.202458                          0.364437   
std                      0.298336                          0.397448   
min                      0.000000                          0.000000   
25%                      0.000000                          0.000000   
50%                      0.083333                          0.166667   
75%                      0.300000                          0.750000   
max                      1.000000                          1.000000   

       CASH_ADVANCE_FREQUENCY  CASH_ADVANCE_TRX  PURCHASES_TRX  CREDIT_LIMIT  \
count             8950.000000       8950.000000    8950.000000   8949.000000   
mean                 0.135144          3.248827      14.709832   4494.449450   
std                  0.200121          6.824647      24.857649   3638.815725   
min                  0.000000          0.000000       0.000000     50.000000   
25%                  0.000000          0.000000       1.000000   1600.000000   
50%                  0.000000          0.000000       7.000000   3000.000000   
75%                  0.222222          4.000000      17.000000   6500.000000   
max                  1.500000        123.000000     358.000000  30000.000000   

           PAYMENTS  MINIMUM_PAYMENTS  PRC_FULL_PAYMENT       TENURE  
count   8950.000000       8637.000000       8950.000000  8950.000000  
mean    1733.143852        864.206542          0.153715    11.517318  
std     2895.063757       2372.446607          0.292499     1.338331  
min        0.000000          0.019163          0.000000     6.000000  
25%      383.276166        169.123707          0.000000    12.000000  
50%      856.901546        312.343947          0.000000    12.000000  
75%     1901.134317        825.485459          0.142857    12.000000  
max    50721.483360      76406.207520          1.000000    12.000000

Bu tablo, yukarıdaki veri setinin özet istatistiklerini içermektedir. Her sütun için ortalama (mean), standart sapma (std), minimum, çeyrek değerler (25%, 50%, 75%) ve maksimum değerleri göstermektedir. Her bir sütun hakkında bazı çıkarımlar şu şekildedir:

1. **BALANCE**: Müşterilerin ortalama kredi kartı bakiyesi 1564.47 TL. Ancak standart sapma yüksek (2081.53), bu da bakiyelerin büyük bir değişkenlik gösterdiğini ifade eder. Bakiyelerin maksimum değeri oldukça yüksek (19043.14 TL).

2. **BALANCE_FREQUENCY**: Ortalama sıklık değeri 0.877'dir; bu da çoğu müşterinin bakiyesini düzenli olarak kontrol ettiğini gösterir. Sıklık değerinin maksimum değeri 1'dir, bu da bakiyesini sürekli güncelleyen müşteriler olduğunu belirtir.

3. **PURCHASES**: Alışveriş harcamalarının ortalaması 1003.20 TL'dir. Ancak, alışveriş miktarlarında yüksek bir değişkenlik var (standart sapma 2136.63), maksimum harcama ise 49039.57 TL gibi çok yüksek bir değer.

4. **ONEOFF_PURCHASES** ve **INSTALLMENTS_PURCHASES**: Tek seferlik ve taksitli alışverişler ortalama olarak sırasıyla 592.44 TL ve 411.07 TL. Ancak, tek seferlik alışverişlerde (maksimum 40761.25 TL) daha geniş bir dağılım görülmekte. Bu, bazı müşterilerin yüksek tutarlı tek seferlik harcamalar yaparken, bazılarının taksitli alışverişi tercih ettiğini gösteriyor.

5. **CASH_ADVANCE**: Ortalama nakit avans 978.87 TL’dir ve dağılımı oldukça geniş (standart sapma 2097.16), maksimum 47137.21 TL’ye kadar çıkabiliyor. Bu durum, bazı müşterilerin yoğun bir şekilde nakit avans kullandığını gösteriyor.

6. **PURCHASES_FREQUENCY**: Alışveriş yapma sıklığı ortalama 0.49, yani yarıya yakın; müşterilerin alışveriş yapma sıklığı geniş bir dağılıma sahip. Maksimum 1 ile bazı müşteriler düzenli alışveriş yaparken, minimum 0 ile hiç alışveriş yapmayan müşteriler de bulunmakta.

7. **CASH_ADVANCE_FREQUENCY** ve **CASH_ADVANCE_TRX**: Nakit avans çekme sıklığı düşük (ortalama 0.135). Çoğu müşteri nadiren nakit avans çekiyor; bu, CASH_ADVANCE_TRX ile de destekleniyor. Nakit avans çekme işlemleri genellikle seyrek (ortalama işlem sayısı 3.24).

8. **CREDIT_LIMIT**: Ortalama kredi limiti 4494.45 TL, ancak bu da geniş bir aralıkta dağılıyor (minimum 50 TL, maksimum 30000 TL). Bu, farklı müşteri segmentlerinde değişik kredi limiti politikalarının uygulanmış olabileceğine işaret ediyor.

9. **PAYMENTS**: Ortalama ödeme miktarı 1733.14 TL ve bazı müşteriler çok yüksek miktarlar ödemiş (maksimum 50721.48 TL).

10. **MINIMUM_PAYMENTS**: Ortalama minimum ödeme miktarı 864.21 TL, ancak maksimum 76406.21 TL gibi aşırı yüksek değerlere ulaşabiliyor. Veri eksiklikleri de mevcut, zira 8637 müşteri için bu veri sağlanmış.

11. **PRC_FULL_PAYMENT**: Müşterilerin yalnızca küçük bir kısmı tam ödeme yapıyor; ortalama 0.15. Tam ödeme yapan müşteriler azınlıkta (maksimum 1, medyan 0).

12. **TENURE**: Kredi kartlarının açık kaldığı süre 6 ila 12 ay arasında değişiyor. Çoğu müşterinin kredi kartı açık kalma süresi 12 aydır.

### Genel Değerlendirme:
Bu özet istatistikler, kredi kartı kullanıcılarının farklı profillere sahip olduğunu gösteriyor. Yüksek standart sapmalar, müşteri davranışlarının oldukça çeşitli olduğunu, bazı müşterilerin sık alışveriş yaparken, bazı müşterilerin nadiren alışveriş yaptığını ya da yoğun bir şekilde nakit avans kullandığını gösteriyor. Özellikle **PURCHASES**, **CASH_ADVANCE** ve **CREDIT_LIMIT** sütunlarındaki büyük değişkenlik, müşteri segmentasyonu ve pazarlama stratejileri geliştirilirken dikkate alınabilir.

In [14]:
# Kredi kartı limiti ve bakiye arasındaki korelasyon
corr_limit_balance = data['CREDIT_LIMIT'].corr(data['BALANCE'])
print("\nKredi Limiti ve Bakiye Korelasyonu:", corr_limit_balance)


Kredi Limiti ve Bakiye Korelasyonu: 0.5312834611329146


In [15]:
# Alışveriş sıklığı ve nakit avans alımı arasındaki korelasyon
corr_freq_cash_advance = data['PURCHASES_FREQUENCY'].corr(data['CASH_ADVANCE'])
print("Alışveriş Sıklığı ve Nakit Avans Korelasyonu:", corr_freq_cash_advance)

Alışveriş Sıklığı ve Nakit Avans Korelasyonu: -0.21550737787778002


In [15]:
# Alışveriş sıklığı ve toplam alışveriş miktarı arasındaki korelasyon
corr_freq_purchases = data['PURCHASES_FREQUENCY'].corr(data['PURCHASES'])
print("Alışveriş Sıklığı ve Toplam Alışveriş Miktarı Korelasyonu:", corr_freq_purchases)


Alışveriş Sıklığı ve Toplam Alışveriş Miktarı Korelasyonu: 0.3930174275339076


In [25]:
# Ödeme oranı ve minimum ödemeler arasındaki korelasyon
corr_payment_min_payments = data['PAYMENTS'].corr(data['MINIMUM_PAYMENTS'])
print("Ödeme Oranı ve Minimum Ödemeler Korelasyonu:", corr_payment_min_payments)


Ödeme Oranı ve Minimum Ödemeler Korelasyonu: 0.12665114274837325


In [18]:
# Kredi limiti ve alışveriş miktarı arasındaki korelasyon
corr_credit_limit_purchase = data['CREDIT_LIMIT'].corr(data['PURCHASES'])
print(" Kredi Limiti ve Alışveriş Miktarı Korelasyonu:", corr_credit_limit_purchase)


 Kredi Limiti ve Alışveriş Miktarı Korelasyonu: 0.35696313640278904


Bu korelasyon değerleri ve gözlemler, kredi kartı kullanıcılarının alışveriş ve ödeme davranışları hakkında çeşitli çıkarımlar yapmamıza olanak tanır:

1. **Alışveriş Sıklığı ve Nakit Avans Korelasyonu (-0.2155)**:
   - Negatif korelasyon (-0.2155), alışveriş sıklığı ile nakit avans çekimi arasında ters bir ilişki olduğunu gösterir. Yani, müşteriler daha sık alışveriş yaptıkça, nakit avans çekme olasılıkları azalır. Bu durum, düzenli alışveriş yapan müşterilerin genellikle nakit avansa daha az ihtiyaç duyduğunu veya alışveriş yapmayı tercih ettiğini düşündürebilir.

2. **Alışveriş Sıklığı ve Toplam Alışveriş Miktarı Korelasyonu (0.3930)**:
   - Pozitif korelasyon (0.3930), müşterilerin daha sık alışveriş yaptıkça toplam harcama miktarlarının da arttığını gösterir. Bu ilişki, sık alışveriş yapan müşterilerin daha yüksek harcama eğiliminde olduğunu doğruluyor. Bu da bu müşterilerin aktif alışveriş davranışına sahip bir müşteri segmenti olarak tanımlanabileceğini gösterir.

3. **Ödeme Oranı ve Minimum Ödemeler Korelasyonu (0.1267)**:
   - Düşük pozitif korelasyon (0.1267), ödeme oranı ile minimum ödemeler arasında zayıf da olsa bir ilişki olduğunu belirtir. Yani minimum ödeme tutarı arttıkça tam ödeme yapma olasılığı biraz artabilir, fakat ilişki çok güçlü değildir. Bu durum, tam ödeme yapmanın sadece minimum ödeme tutarından etkilenmediğini, başka faktörlerin de devrede olabileceğini gösterir.

4. **Kredi Limiti ve Alışveriş Miktarı Korelasyonu (0.3570)**:
   - Pozitif korelasyon (0.3570), kredi limiti ile toplam alışveriş miktarı arasında anlamlı bir ilişki olduğunu gösterir. Kredi limiti yüksek olan müşteriler, daha yüksek alışveriş harcaması yapma eğilimindedir. Kredi limitlerinin müşteri harcama kapasitesini arttırıcı bir etkisi olduğu söylenebilir.


### Genel Yorumlar:
Bu korelasyon analizleri, kredi kartı kullanıcılarının finansal davranışlarını anlamak için kullanılabilecek değerli içgörüler sunar:
- **Nakit Avans ve Alışveriş Tercihleri**: Alışverişe odaklanan müşterilerin nakit avansa daha az yönelmesi, ürün veya kampanyalar yoluyla müşterileri alışverişe teşvik ederek nakit avans talebini azaltma stratejisi geliştirebilir.
- **Kredi Limiti ve Harcama**: Kredi limiti arttıkça harcamanın artması, kredi kartı limit artırımının harcama teşvikinde etkili olabileceğini gösterir.
- **Tam Ödeme Davranışı**: Tam ödeme ile minimum ödeme arasındaki zayıf ilişki, ödeme alışkanlıklarının ve müşteri likiditesinin birçok faktöre bağlı olduğunu düşündürmektedir.

In [18]:
# En yüksek tek seferlik alışveriş miktarı
max_oneoff_purchase = data['ONEOFF_PURCHASES'].max()
print("\nEn Yüksek Tek Seferlik Alışveriş Miktarı:", max_oneoff_purchase)


En Yüksek Tek Seferlik Alışveriş Miktarı: 40761.25


In [19]:
# En çok yapılan alışveriş türü
most_common_purchase_type = data[['ONEOFF_PURCHASES', 'INSTALLMENTS_PURCHASES']].idxmax(axis=1).value_counts().idxmax()
print("En Çok Yapılan Alışveriş Türü:", most_common_purchase_type)

En Çok Yapılan Alışveriş Türü: ONEOFF_PURCHASES


In [20]:
# En uzun süreli müşteri
oldest_customer = data.loc[data['TENURE'].idxmax(), 'CUST_ID']
print("En Uzun Süreli Müşteri:", oldest_customer)

En Uzun Süreli Müşteri: C10001


In [21]:
# Nakit avans kullanımı ve toplam alışveriş miktarı arasındaki fark
data['Cash_vs_Purchases'] = data['CASH_ADVANCE'] - data['PURCHASES']
mean_diff_cash_purchases = data['Cash_vs_Purchases'].mean()
print("\nNakit Avans Kullanımı ve Toplam Alışveriş Miktarı Farkının Ortalaması:", mean_diff_cash_purchases)



Nakit Avans Kullanımı ve Toplam Alışveriş Miktarı Farkının Ortalaması: -24.33372105407822


In [23]:
# Ödeme yapma oranı ve kredi kartı limiti arasındaki fark
data['Payment_vs_Limit'] = data['PAYMENTS'] - data['CREDIT_LIMIT']
mean_diff_payment_limit = data['Payment_vs_Limit'].mean()
print("Ödemeler ve Kredi Kartı Limiti Farkının Ortalaması:", mean_diff_payment_limit)


Ödemeler ve Kredi Kartı Limiti Farkının Ortalaması: -2761.1129395136886


In [26]:
# Alışveriş yapma sıklığı ve kredi limiti arasındaki fark
data['Freq_vs_Limit'] = data['PURCHASES_FREQUENCY'] - data['CREDIT_LIMIT']
mean_diff_freq_limit = data['Freq_vs_Limit'].mean()
print("Alışveriş Sıklığı ve Kredi Kartı Limiti Farkının Ortalaması:", mean_diff_freq_limit)

Alışveriş Sıklığı ve Kredi Kartı Limiti Farkının Ortalaması: -4493.959045022348


In [30]:
# Alışveriş yapan müşteri sayısı
total_purchasers = data[data['PURCHASES'] > 0]['CUST_ID'].nunique()
print(" Alışveriş Yapan Müşteri Sayısı:", total_purchasers)


 Alışveriş Yapan Müşteri Sayısı: 6906


In [29]:
# Ortalama alışveriş tutarı
avg_purchase_amount = data['PURCHASES'].mean()
print(" Ortalama Alışveriş Tutarı:", avg_purchase_amount)


 Ortalama Alışveriş Tutarı: 1003.2048335195531


In [31]:
# En yüksek kredi limitine sahip müşteri
max_credit_limit_cust = data.loc[data['CREDIT_LIMIT'].idxmax(), 'CUST_ID']
print("En Yüksek Kredi Limitine Sahip Müşteri:", max_credit_limit_cust)


En Yüksek Kredi Limitine Sahip Müşteri: C15041


In [32]:
# En yüksek nakit avans veren müşteri
max_cash_advance_cust = data.loc[data['CASH_ADVANCE'].idxmax(), 'CUST_ID']
print("En Yüksek Nakit Avans Veren Müşteri:", max_cash_advance_cust)


En Yüksek Nakit Avans Veren Müşteri: C12226


In [33]:
# Ortalama ödeme miktarı
avg_payment_amount = data['PAYMENTS'].mean()
print("Ortalama Ödeme Miktarı:", avg_payment_amount)


Ortalama Ödeme Miktarı: 1733.1438520248046


In [34]:
# En sadık müşteriler (En yüksek TENURE'e sahip)
most_tenured_custs = data.nlargest(5, 'TENURE')['CUST_ID'].tolist()
print("En Sadık Müşteriler (En Yüksek TENURE):", most_tenured_custs)


En Sadık Müşteriler (En Yüksek TENURE): ['C10001', 'C10002', 'C10003', 'C10004', 'C10005']


In [35]:
# En yüksek ödeme yapan müşteri
max_payment_cust = data.loc[data['PAYMENTS'].idxmax(), 'CUST_ID']
print("En Yüksek Ödeme Yapan Müşteri:", max_payment_cust)


En Yüksek Ödeme Yapan Müşteri: C14338


In [36]:
#  Alışveriş türleri (Tek seferlik ve taksitli) arasındaki fark
data['Oneoff_vs_Installments'] = data['ONEOFF_PURCHASES'] - data['INSTALLMENTS_PURCHASES']
mean_diff_oneoff_installments = data['Oneoff_vs_Installments'].mean()
print("Alışveriş Türleri (Tek Seferlik ve Taksitli) Arasındaki Farkın Ortalaması:", mean_diff_oneoff_installments)


Alışveriş Türleri (Tek Seferlik ve Taksitli) Arasındaki Farkın Ortalaması: 181.36972625698326


Bu veriler, müşterilerin kredi kartı kullanım alışkanlıkları ve finansal davranışları hakkında daha detaylı içgörüler sağlar. Her bir maddeyi inceleyelim:

1. **En Yüksek Tek Seferlik Alışveriş Miktarı: 40761.25 TL**:
   - Bu oldukça yüksek bir harcama, bazı müşterilerin yüksek fiyatlı ürün veya hizmetler için kredi kartlarını yoğun bir şekilde kullanabildiğini gösteriyor. Bu tür harcamalar, lüks harcamalar veya beklenmedik büyük ödemeler için yapılmış olabilir.

2. **En Çok Yapılan Alışveriş Türü: ONEOFF_PURCHASES (Tek Seferlik Alışverişler)**:
   - Tek seferlik alışverişler, bu veri setinde en çok tercih edilen alışveriş türüdür. Bu durum, birçok müşterinin kredi kartını taksitli harcamalar yerine tek seferlik harcamalar için tercih ettiğini gösterir. Bu tür harcamalar, kredi kartı kullanım alışkanlıklarının kısa vadeli ihtiyaçları karşılamaya yönelik olduğunu düşündürebilir.

3. **En Uzun Süreli Müşteri: C10001**:
   - C10001 kodlu müşteri, kredi kartını en uzun süredir aktif olarak kullanan müşteri olarak dikkat çekiyor. Bu müşteri, kredi kartı şirketi için potansiyel olarak sadık bir müşteri olabilir. Kredi kartını uzun süre kullanan müşteriler, genellikle daha güvenilir kabul edilir.

4. **Nakit Avans Kullanımı ve Toplam Alışveriş Miktarı Farkının Ortalaması: -24.33 TL**:
   - Bu farkın negatif olması, müşterilerin alışveriş miktarlarının genelde nakit avans kullanım miktarından daha yüksek olduğunu gösterir. Bu durum, müşterilerin nakit avans yerine alışverişe öncelik verdiğini veya nakit avans kullanımının daha az tercih edildiğini gösterebilir.

5. **Ödemeler ve Kredi Kartı Limiti Farkının Ortalaması: -2761.11 TL**:
   - Ortalama olarak müşterilerin ödemeleri, kredi kartı limitlerinden daha düşük kalmaktadır. Bu, müşterilerin kredi limitlerini tamamen kullanma eğiliminde olmadıklarını veya borçlarını düzenli olarak tam kapatmadıklarını gösterebilir.

6. **Alışveriş Sıklığı ve Kredi Kartı Limiti Farkının Ortalaması: -4493.96 TL**:
   - Negatif fark, alışveriş sıklığının kredi kartı limiti ile doğrudan ilişkili olmadığını ve kredi limitinin yüksekliğine rağmen müşterilerin sık alışveriş yapma oranının düşük olduğunu düşündürebilir. Bu durum, müşterilerin limitleri yüksek olsa bile harcamalarını kısıtlayabileceklerini gösterir.

7. **Alışveriş Yapan Müşteri Sayısı: 6906**:
   - Bu veri, toplam müşteri sayısının yaklaşık %77’sinin aktif olarak alışveriş yaptığını gösteriyor. Bu yüksek oran, kredi kartlarının alışveriş için aktif olarak kullanıldığını göstermektedir.

8. **Ortalama Alışveriş Tutarı: 1003.20 TL**:
   - Ortalama alışveriş tutarının 1000 TL civarında olması, müşterilerin genellikle orta seviyede alışveriş yaptığını gösterir. Bu bilgi, harcama eğilimlerinin anlaşılması için değerlidir.

9. **En Yüksek Kredi Limitine Sahip Müşteri: C15041**:
   - C15041 kodlu müşteri, en yüksek kredi limitine sahip müşteri olarak dikkat çekiyor. Bu, müşterinin kredi risk değerlendirmesinde güvenilir bulunduğunu veya büyük harcamalar yapma eğiliminde olduğunu gösterebilir.

10. **En Yüksek Nakit Avans Veren Müşteri: C12226**:
    - C12226 kodlu müşteri, en yüksek nakit avans kullanımına sahip. Bu müşteri, acil nakit ihtiyacı olan veya nakit avansı bir alışkanlık haline getiren biri olabilir.

11. **Ortalama Ödeme Miktarı: 1733.14 TL**:
    - Ortalama ödeme miktarı, müşterilerin düzenli ödeme yaptığını ancak limitlerini tamamen kapatmadığını gösteriyor. Bu ortalama, minimum ödemeleri aşan ancak tam kapatmaya ulaşmayan bir ödeme düzenini gösteriyor.

12. **En Sadık Müşteriler (En Yüksek TENURE)**:
    - En uzun süreli müşteriler listesi (C10001, C10002, C10003, C10004, C10005) müşteri sadakati açısından dikkat çekicidir. Bu müşteriler, potansiyel olarak daha güvenilir kabul edilebilir ve sadakat programları ile ödüllendirilebilir.

13. **En Yüksek Ödeme Yapan Müşteri: C14338**:
    - C14338 kodlu müşteri, en yüksek ödeme yapan müşteri olarak öne çıkıyor. Bu müşteri, düzenli ve yüksek ödemeler yapma eğiliminde olduğundan, kredi kartı şirketi için değerli bir müşteri olabilir.

14. **Alışveriş Türleri (Tek Seferlik ve Taksitli) Arasındaki Farkın Ortalaması: 181.37 TL**:
    - Bu pozitif fark, müşterilerin tek seferlik alışverişlere taksitli alışverişlerden ortalama 181 TL daha fazla harcadığını gösterir. Müşteriler genellikle büyük tutarlarda taksitli alışveriş yapmayı tercih etmekle birlikte, tek seferlik alışverişler daha sık ve daha yüksek tutarda gerçekleşiyor olabilir.

### Genel Değerlendirme:
Bu veriler, kredi kartı kullanım alışkanlıkları, ödeme eğilimleri, alışveriş türleri ve müşteri sadakati hakkında detaylı içgörüler sağlar. Özellikle **tek seferlik alışverişlerin yoğunluğu** ve **kredi limiti ile alışveriş ilişkisi**, kredi kartı hizmetlerinin optimize edilmesi ve kişiselleştirilmiş kampanyalar oluşturulması için kullanılabilir.

In [39]:
# Kullanıcıların kredi kartı limiti ile yaptığı alışveriş miktarı arasındaki ilişkiyi incele
data['Utilization_Rate'] = data['PURCHASES'] / data['CREDIT_LIMIT']

# Kullanıcıların alışveriş sıklıkları üzerinde bir analiz yapın
purchase_freq_analysis = pd.cut(data['PURCHASES_FREQUENCY'], bins=[0, 0.25, 0.5, 0.75, 1],
                                labels=['Düşük', 'Orta', 'Yüksek', 'Çok Yüksek']).value_counts()
print("Alışveriş Frekansı Analizi:")
print(purchase_freq_analysis)

Alışveriş Frekansı Analizi:
PURCHASES_FREQUENCY
Çok Yüksek    3104
Düşük         1607
Orta          1153
Yüksek        1043
Name: count, dtype: int64


In [40]:
# Kullanıcıların kredi kartı limitleri ile yaptıkları alışveriş miktarı arasındaki ilişkiyi incele
utilization_rate_analysis = pd.cut(data['Utilization_Rate'], bins=[0, 0.2, 0.4, 0.6, 0.8, 1],
                                   labels=['0-20%', '20-40%', '40-60%', '60-80%', '80-100%']).value_counts()
print("\nKullanım Oranı Analizi:")
print(utilization_rate_analysis)


Kullanım Oranı Analizi:
Utilization_Rate
0-20%      3533
20-40%     1486
40-60%      782
60-80%      415
80-100%     249
Name: count, dtype: int64


In [41]:
# Ödeme alışkanlıkları üzerinde bir analiz yapın
data['Payment_Ratio'] = data['MINIMUM_PAYMENTS'] / data['PAYMENTS']

payment_ratio_analysis = pd.cut(data['Payment_Ratio'], bins=[0, 0.25, 0.5, 0.75, 1],
                                labels=['Düşük', 'Orta', 'Yüksek', 'Çok Yüksek']).value_counts()
print("\nÖdeme Oranı Analizi:")
print(payment_ratio_analysis)


Ödeme Oranı Analizi:
Payment_Ratio
Düşük         3095
Orta          1392
Çok Yüksek     946
Yüksek         839
Name: count, dtype: int64


Bu analizler, kullanıcıların kredi kartı kullanımı, kredi limitleri ve ödeme alışkanlıkları hakkında çok katmanlı bir anlayış sunuyor. Her bir analizden çıkarılabilecek temel sonuçlara bakalım:

### 1. Alışveriş Frekansı Analizi
- **Çok Yüksek (3104)**: Kullanıcıların en büyük kısmı çok yüksek alışveriş frekansına sahip, bu da kredi kartlarının alışveriş için yoğun bir şekilde kullanıldığını gösteriyor. Bu grup, muhtemelen düzenli olarak kredi kartını tercih eden, alışveriş odaklı kullanıcıları temsil eder.
- **Düşük (1607), Orta (1153), Yüksek (1043)**: Orta ve düşük alışveriş frekansına sahip kullanıcı sayısı, yüksek frekanslı kullanıcı sayısından fazla. Bu durum, müşteri segmentinin bir kısmının kredi kartını yalnızca belirli zamanlarda veya özel durumlar için kullandığını gösteriyor.

Bu frekans dağılımı, **kredi kartı kampanyalarını farklı segmentlere göre özelleştirme** fırsatı sunuyor. Örneğin, düşük frekanstaki kullanıcılar için daha fazla harcama yapmaya teşvik edici kampanyalar geliştirilebilir.

### 2. Kullanım Oranı Analizi (Kredi Kartı Limiti - Alışveriş Miktarı)
- **0-20% (3533)**: En büyük grup, kredi limitinin yalnızca %20’sine kadarını kullanan kullanıcılar. Bu, müşterilerin çoğunlukla düşük oranlarda limit kullanımını tercih ettiğini gösteriyor. Bu kullanıcıların kredi kartlarını daha az riskli ve daha yönetilebilir seviyelerde kullandıkları söylenebilir.
- **20-40% (1486) ve 40-60% (782)**: Bu dilimlerde de kullanıcı sayısı önemli bir düzeydedir ve kullanıcıların kredi limitlerinin %20 ila %60'ını kullandıklarını gösterir.
- **60-100% Arası (664)**: Bu kullanıcılar kredi limitinin büyük bir kısmını veya tamamını kullanıyorlar. Bu durum, bu müşterilerin nakit ihtiyaçlarının yüksek olduğunu veya limitlerini aktif olarak harcama için kullandıklarını düşündürebilir.

**Kredi kartı limitini düşük oranlarda kullanan** müşterilere, limit artırımı gibi fırsatlar sunulabilirken, limitlerini yüksek oranlarda kullanan kullanıcılar için ödeme kolaylıkları veya kampanyalar oluşturmak faydalı olabilir.

### 3. Ödeme Alışkanlıkları Analizi
- **Düşük Ödeme Oranı (3095)**: Çoğunluk düşük ödeme oranına sahip, yani bu grup genellikle borcunun küçük bir kısmını veya yalnızca minimum ödemesini yapıyor. Bu, kullanıcıların bir kısmının borçlarını uzun vadeye yaymayı veya tam ödeme yapmamayı tercih ettiğini gösterir.
- **Orta ve Yüksek Ödeme Oranı (1392 ve 839)**: Bu gruplar borçlarını daha düzenli ve daha büyük miktarlarda kapatan kullanıcıları temsil eder. Bu müşterilerin finansal durumları daha güçlü olabilir ve ödemelerini düzenli olarak yapma eğilimindedir.
- **Çok Yüksek Ödeme Oranı (946)**: Bu grup, borçlarının büyük kısmını ya da tamamını ödeyen kullanıcıları içerir. Bu, kredi kartını yalnızca kısa vadeli finansal ihtiyaçlar için kullanan, borçlarını ise hızlıca kapatan sadık müşterileri gösterir.

Bu analiz, **ödeme alışkanlıklarına göre kişiselleştirilmiş ödeme planları veya ödül programları** oluşturmak için kullanılabilir. Örneğin, düşük ödeme oranına sahip müşteriler için taksitlendirme veya faiz indirimi gibi kampanyalar, daha yüksek ödeme oranına sahip müşteriler için ise farklı avantajlar sunulabilir.

### Genel Değerlendirme:
Bu üç analiz, kredi kartı kullanıcılarının alışveriş, kredi limit kullanımı ve ödeme alışkanlıkları gibi farklı özelliklerine dayalı olarak **farklı müşteri segmentleri** oluşturulmasını sağlar. Her segment, kendi finansal davranışına göre hedeflenmiş teklifler veya kampanyalarla desteklenebilir. Bu da, müşteri bağlılığını artırmanın yanı sıra müşteri ihtiyaçlarına daha iyi cevap vermek adına önemli bir stratejik avantaj sağlar.

In [60]:

# Alışveriş türlerine göre müşteri segmentasyonu
data['Purchase_Type'] = 'None'
data.loc[(data['ONEOFF_PURCHASES'] > 0) & (data['INSTALLMENTS_PURCHASES'] == 0), 'Purchase_Type'] = 'One-off'
data.loc[(data['ONEOFF_PURCHASES'] == 0) & (data['INSTALLMENTS_PURCHASES'] > 0), 'Purchase_Type'] = 'Installments'
data.loc[(data['ONEOFF_PURCHASES'] > 0) & (data['INSTALLMENTS_PURCHASES'] > 0), 'Purchase_Type'] = 'Both'

# Alışveriş türlerine göre gruplama ve toplam alışveriş miktarını hesaplama
purchase_type_analysis = data.groupby('Purchase_Type')['PURCHASES'].sum()

# En yüksek kredi kartı limitine sahip müşterilerin alışveriş sıklığı ortalaması
max_credit_limit_custs = data.nlargest(100, 'CREDIT_LIMIT')
mean_purchase_freq_max_limit = max_credit_limit_custs['PURCHASES_FREQUENCY'].mean()

# Kredi kartı limiti ile alışveriş sıklığı arasındaki korelasyon
corr_limit_freq = data['CREDIT_LIMIT'].corr(data['PURCHASES_FREQUENCY'])

# En uzun süreli müşterilerin alışveriş türlerine göre dağılımı
oldest_custs = data.nlargest(100, 'TENURE')['Purchase_Type'].value_counts()

# Alışveriş türlerine göre ortalama kredi kartı limiti
mean_credit_limit_purchase_type = data.groupby('Purchase_Type')['CREDIT_LIMIT'].mean()

# Sonuçları yazdırma
print("Alışveriş Türlerine Göre Toplam Alışveriş Miktarı:")
print(purchase_type_analysis)
print("\nEn Yüksek Kredi Kartı Limitine Sahip Müşterilerin Alışveriş Sıklığı Ortalaması:", mean_purchase_freq_max_limit)
print("\nKredi Kartı Limiti ile Alışveriş Sıklığı Korelasyonu:", corr_limit_freq)
print("\nEn Uzun Süreli Müşterilerin Alışveriş Türlerine Göre Dağılımı:")
print(oldest_custs)
print("\nAlışveriş Türlerine Göre Ortalama Kredi Kartı Limiti:")
print(mean_credit_limit_purchase_type)


Alışveriş Türlerine Göre Toplam Alışveriş Miktarı:
Purchase_Type
Both            6290893.31
Installments    1213403.53
None                  0.00
One-off         1474386.42
Name: PURCHASES, dtype: float64

En Yüksek Kredi Kartı Limitine Sahip Müşterilerin Alışveriş Sıklığı Ortalaması: 0.68819696

Kredi Kartı Limiti ile Alışveriş Sıklığı Korelasyonu: 0.11978815438172262

En Uzun Süreli Müşterilerin Alışveriş Türlerine Göre Dağılımı:
Purchase_Type
Both            33
One-off         27
Installments    23
None            17
Name: count, dtype: int64

Alışveriş Türlerine Göre Ortalama Kredi Kartı Limiti:
Purchase_Type
Both            5738.829463
Installments    3371.864329
None            4026.490283
One-off         4515.920572
Name: CREDIT_LIMIT, dtype: float64


Bu analizler, kullanıcıların kredi kartı limitleri, alışveriş türleri, ve alışveriş sıklıkları gibi faktörler arasındaki ilişkiyi daha ayrıntılı olarak anlamamıza yardımcı oluyor. Her bir başlık altında detaylara bakalım:

### 1. Alışveriş Türlerine Göre Toplam Alışveriş Miktarı
- **"Both" (Hem tek seferlik hem de taksitli alışveriş)** türü toplamda 6.290.893,31 TL ile en yüksek alışveriş tutarına sahip. Bu, birçok kullanıcının hem tek seferlik hem de taksitli harcamaları bir arada tercih ettiğini gösteriyor.
- **Taksitli Alışverişler (Installments):** Toplamda 1.213.403,53 TL harcanmış. Bazı kullanıcıların büyük miktarlarda harcamalar yaparak ödemelerini taksitlere bölmeyi tercih ettiklerini gösteriyor.
- **Tek Seferlik Alışverişler (One-off):** 1.474.386,42 TL ile ikinci sırada yer alıyor ve taksitli harcamalardan biraz daha yüksek. Bu durum, birçok müşterinin kredi kartını tek seferlik ve daha kısa vadeli alışverişler için de yoğun şekilde kullandığını gösteriyor.
  
Bu dağılım, kredi kartı şirketlerine, **müşterilerin alışveriş türü tercihleri doğrultusunda kampanyalar geliştirmede** yol gösterebilir. Örneğin, hem taksitli hem de tek seferlik alışveriş yapan müşterilere özel fırsatlar sunulabilir.

### 2. En Yüksek Kredi Kartı Limitine Sahip Müşterilerin Alışveriş Sıklığı Ortalaması: 0.688
- Yüksek kredi kartı limitine sahip müşterilerin alışveriş sıklığı ortalama 0,688 ile oldukça aktif. Bu, yüksek limite sahip müşterilerin alışverişi daha sık yaptığını ve kredi kartını düzenli kullandığını gösterir. Bu müşterilere **sadakat programları veya kredi kartı limitine dayalı ödüller** sunmak, müşteri memnuniyetini artırabilir.

### 3. Kredi Kartı Limiti ile Alışveriş Sıklığı Korelasyonu: 0.12
- Kredi kartı limiti ile alışveriş sıklığı arasındaki korelasyon, düşük ve pozitif bir ilişkiye işaret ediyor. Yani limit arttıkça alışveriş sıklığı çok az da olsa artıyor. Bu düşük korelasyon, kullanıcıların kredi limitinin yüksek olması durumunda bile alışveriş sıklığını aynı seviyelerde koruduğunu düşündürüyor. Yüksek limitli müşteriler, **alışveriş sıklıklarını arttırmaları için teşvik edilebilir**.

### 4. En Uzun Süreli Müşterilerin Alışveriş Türlerine Göre Dağılımı
- **Hem tek seferlik hem de taksitli alışveriş** türünü tercih eden uzun süreli müşterilerin yuzdesi 33 ile en yüksek. Bu, uzun süreli müşterilerin geniş bir harcama yelpazesi içinde daha esnek bir kullanım tercih ettiğini gösteriyor.
- **Tek Seferlik Alışveriş (One-off)** tercih eden uzun süreli müşteri yuzdesi 27 iken, sadece **taksitli alışveriş (Installments)** tercih edenlerin yuzdesi 23.
- **Alışveriş yapmayanlar (None)** yuzdesi 17  ile en düşük grupta yer alıyor. Bu müşteriler, kredi kartlarını daha çok limit sağlamak için aktif tutan ancak alışveriş için kullanmayan veya nadir kullanan kişiler olabilir.

Bu analiz, **uzun süreli müşterilerin kredi kartı kullanım alışkanlıklarının daha çeşitli olduğunu ve farklı alışveriş türlerine hitap eden kampanyalar ile desteklenebileceğini** gösteriyor.

### 5. Alışveriş Türlerine Göre Ortalama Kredi Kartı Limiti
- **Hem tek seferlik hem de taksitli alışveriş yapanlar ("Both")** grubunda ortalama kredi limiti 5.738 TL ile en yüksek. Bu, yüksek limitli müşterilerin daha geniş harcama seçeneklerinden faydalandığını gösterir.
- **Taksitli alışveriş (Installments)** tercih edenlerde ise ortalama kredi limiti 3.371 TL ile daha düşük. Bu kullanıcılar, genellikle tek seferde yüksek tutarda alışveriş yerine küçük ödemelerle uzun vadeli ödeme planlarını tercih edebilirler.
- **Tek seferlik alışveriş (One-off)** yapanlar için kredi limiti ortalama 4.515 TL ile orta seviyelerde. Bu kullanıcılar tek seferde büyük harcamalar yapıyor olabilir, ancak limiti artırmak gibi bir eğilim göstermiyor olabilirler.

Bu bulgular, **alışveriş türlerine göre kredi limiti düzenleme** veya **kampanya geliştirme stratejilerinde** yol gösterici olabilir. Örneğin, hem tek seferlik hem taksitli harcama yapan müşterilere daha yüksek kredi limiti ve farklı alışveriş avantajları sunmak, müşteri memnuniyeti ve bağlılığını artırabilir.

### Genel Değerlendirme
Bu analizler, kredi kartı kullanıcılarının alışveriş türleri, kredi limitleri ve alışveriş sıklıkları gibi faktörlerin birbirine nasıl etki ettiğini gösteriyor. Özellikle "Both" kategorisinde yer alan müşteriler, daha yüksek limit ve sık alışveriş yapma eğiliminde olduklarından, şirket için değerli bir müşteri grubu oluşturabilir. Bu müşteriler için özel teklifler veya yüksek limitler sunmak, kredi kartı kullanımını daha da artırabilir.

In [12]:
import pandas as pd
from sklearn.cluster import KMeans
from sklearn.preprocessing import StandardScaler

# Kullanılacak özellikler
features = ['BALANCE', 'PURCHASES', 'ONEOFF_PURCHASES', 
            'INSTALLMENTS_PURCHASES', 'CASH_ADVANCE', 
            'CREDIT_LIMIT', 'PAYMENTS', 'MINIMUM_PAYMENTS']

# 1. Eksik değerleri doldurma
data.fillna(method='ffill', inplace=True)

# 2. Veri normalizasyonu
scaler = StandardScaler()
scaled_data = scaler.fit_transform(data[features])

# 3. KMeans modelini tanımlama ve eğitme
kmeans = KMeans(n_clusters=4, random_state=42, n_init=10)  # n_init parametresi eklenmiştir
data['cluster'] = kmeans.fit_predict(scaled_data)

# 4. Her küme için istatistiksel özet alma
cluster_summary = data.groupby('cluster')[features].describe()

# 5. Her küme için ortalamaları alma
cluster_means = data.groupby('cluster')[features].mean()

# 6. Her küme için en iyi ve en kötü özellikleri bulma
best_features = cluster_means.idxmax(axis=1)
worst_features = cluster_means.idxmin(axis=1)

# 7. Sonuçları yazdırma
for cluster in cluster_means.index:
    best = best_features[cluster]
    worst = worst_features[cluster]
    print(f"Cluster {cluster}: En iyi özellik - {best}, En kötü özellik - {worst}")


Cluster 0: En iyi özellik - ('CREDIT_LIMIT', 'mean'), En kötü özellik - ('INSTALLMENTS_PURCHASES', 'mean')
Cluster 1: En iyi özellik - ('CREDIT_LIMIT', 'mean'), En kötü özellik - ('CASH_ADVANCE', 'mean')
Cluster 2: En iyi özellik - ('PAYMENTS', 'mean'), En kötü özellik - ('CASH_ADVANCE', 'mean')
Cluster 3: En iyi özellik - ('CREDIT_LIMIT', 'mean'), En kötü özellik - ('INSTALLMENTS_PURCHASES', 'mean')


Bu analiz, müşterilerin farklı kümelere ayrıldığını ve her kümenin belirli özelliklere göre sınıflandırıldığını gösteriyor. Her bir kümenin en iyi ve en kötü özelliklerine göz atalım:

### 1. Cluster 0
- **En İyi Özellik - Kredi Limiti (CREDIT_LIMIT, mean):** Bu kümeye ait müşterilerin kredi limitleri yüksek. Bu, kredi limitlerinin avantajlı kullanımıyla öne çıkan bir müşteri grubu olduğunu gösteriyor. Bu müşterilerin genellikle finansal güvenilirlikleri yüksek ve kredi kartı kullanımları düzenli olabilir.
- **En Kötü Özellik - Taksitli Alışveriş (INSTALLMENTS_PURCHASES, mean):** Bu kümedeki müşteriler, taksitli alışverişleri çok tercih etmiyor veya daha az miktarda yapıyor. Bu, kullanıcıların tek seferlik veya nakit harcamalara daha fazla odaklandığını düşündürebilir.

**Genel Yorum:** Cluster 0, yüksek kredi limitine sahip olan ancak taksitli alışverişe yönelmeyen bir müşteri segmentini temsil ediyor. Bu kümeye yönelik kampanyalarda taksitli alışveriş avantajları sunulabilir.

### 2. Cluster 1
- **En İyi Özellik - Kredi Limiti (CREDIT_LIMIT, mean):** Cluster 1’deki müşterilerin de kredi limitleri yüksek. Bu müşteriler, finansal olanakları güçlü olan bir grubu oluşturuyor.
- **En Kötü Özellik - Nakit Avans (CASH_ADVANCE, mean):** Bu kümeye ait müşteriler, nakit avansı daha az kullanıyor. Yani, günlük veya acil nakit ihtiyaçları için kredi kartını kullanmak yerine, alışveriş için daha fazla odaklanıyor olabilirler.

**Genel Yorum:** Cluster 1, kredi kartını alışveriş odaklı kullanan ancak nakit avans kullanımını tercih etmeyen bir müşteri grubu olarak öne çıkıyor. Bu müşteriler için nakit avans çekimlerinde avantaj sağlanabilir veya kredi kartı limiti arttırılarak alışveriş teşvik edilebilir.

### 3. Cluster 2
- **En İyi Özellik - Ödemeler (PAYMENTS, mean):** Cluster 2 müşterileri, ödemelerini düzenli ve yüksek tutarlarda yapan bir segmenti oluşturuyor. Bu durum, finansal yükümlülüklerini zamanında yerine getiren ve borçlarını kapatmaya yönelik bir eğilim sergileyen güvenilir müşterilere işaret ediyor.
- **En Kötü Özellik - Nakit Avans (CASH_ADVANCE, mean):** Bu küme de nakit avansı çok tercih etmiyor. Bu durum, bu grubun yalnızca alışveriş için kredi kartını kullanma eğiliminde olduğunu düşündürebilir.

**Genel Yorum:** Cluster 2, ödemelerini düzenli yapan ancak nakit avans ihtiyacı olmayan, güvenilir bir müşteri grubu. Bu müşterilere, ödeme avantajları veya geri ödeme kolaylıkları sunulabilir.

### 4. Cluster 3
- **En İyi Özellik - Kredi Limiti (CREDIT_LIMIT, mean):** Bu kümedeki müşteriler de yüksek kredi limitine sahip. Diğer kümelerde olduğu gibi, bu kümedeki müşteriler de finansal açıdan güçlü bir grup olarak değerlendirilebilir.
- **En Kötü Özellik - Taksitli Alışveriş (INSTALLMENTS_PURCHASES, mean):** Cluster 3 müşterileri taksitli alışverişi düşük seviyede yapıyor, yani bu müşteriler de daha çok tek seferlik veya farklı harcama türlerine odaklanıyor olabilir.

**Genel Yorum:** Cluster 3, yüksek kredi limitine sahip ancak taksitli alışveriş yerine diğer alışveriş türlerini tercih eden müşterileri kapsıyor. Bu küme için taksitli alışveriş kampanyaları, harcama alışkanlıklarını artırmak adına kullanılabilir.

### Genel Değerlendirme
Bu analiz, **müşteri kümelerinin her birinin farklı finansal alışkanlıklara** sahip olduğunu ve kredi kartı kullanımında farklı stratejiler geliştirdiğini gösteriyor. Kredi kartı limitini avantajlı kullananlar, taksitli alışverişi düşük olanlar, nakit avans kullanmayanlar ve düzenli ödeme yapanlar gibi segmentler mevcut. Bu kümelere özel olarak:
- **Taksitli alışveriş kampanyaları, nakit avans teşvikleri veya ödüllü ödeme programları** gibi farklı pazarlama stratejileri geliştirilebilir. 
- Ayrıca, her kümenin özelliklerine göre sunulacak finansal ürünler, müşteri memnuniyetini ve kredi kartı kullanımını artırabilir.